tests/ui/lazy_lock_like/auxiliary/once_cell.rs

```diff
@@ -0,0 +1,52 @@
+//! **FAKE** once_cell crate.
+
+pub mod sync {
+    use core::cell::Cell;
+    use std::marker::PhantomData;
+
+    pub struct OnceCell<T>(PhantomData<T>);
+    impl<T> OnceCell<T> {
+        pub const fn new() -> OnceCell<T> {
+            OnceCell(PhantomData)
+        }
+    }
+    impl<T> Default for OnceCell<T> {
+        fn default() -> Self {
+            Self::new()
+        }
+    }
+
+    pub struct Lazy<T, F = fn() -> T> {
+        cell: OnceCell<T>,
+        init: Cell<Option<F>>,
+    }
+    unsafe impl<T, F: Send> Sync for Lazy<T, F> where OnceCell<T>: Sync {}
+    impl<T, F> Lazy<T, F> {
+        pub const fn new(f: F) -> Lazy<T, F> {
+            Lazy {
```

Comment:
It works nicely with other functions but not this `new` function unfortunately~ otherwise it reports error in my test code where saying 'could not evaluate static initializer'

Reply:
You could try just `panic!()` instead.

Reply:
same, it couldn't pass const evaluation I think

Reply:
Oh oh oh duh `new` is actually called at compile time. Don't know what I was thinking.